### PR TITLE
icu: 73.2 -> 74.2

### PR DESCRIPTION
**Pull request by afh**
https://icu.unicode.org/download/74

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by afh** [comment]
Obsoleted by #290266

**Comment by 7c6f434c** [comment]
Obsoleted in what sense??

**Comment by afh** [comment]
My sincerest apologies, I must've had a complete 🧠😶‍🌫️ earlier today.

**Comment by marsam** [comment]
@GrahamcOfBorg eval

**Comment by afh** [comment]
Friendly ping to @mweinelt, @AndersonTorres, @anthonyroussel on this 🙂

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -21447,7 +21447,7 @@ with pkgs;
     icu74
   ;
 
-  icu = icu73;
+  icu = icu74;
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:21450]: A mere change of defaults? Well, let's wait ofBorg then.

Comment (afh) [reply pkgs/top-level/all-packages.nix:21450]: Seems like all checks [except for tests on x86_64-darwin](https://github.com/NixOS/nixpkgs/pull/290509/checks?check_run_id=25692459832) have passed, not sure if these are related.

**Comment by 7c6f434c** [comment]
Darwin failure is LLVM timeout on staging (which is a stdenv dependency), this is not specific to this PR (and there is some work being done on ofBorg side to find a proper solution to this in the medium term)

**Comment by 7c6f434c** [comment]
Note that ofBorg does not do a full reverse-dependency build so maybe if someone can guess possibly-fragile packages for ICU version bump we should ofborg-build them explicitly.

**Comment by AndersonTorres** [comment]
> Note that ofBorg does not do a full reverse-dependency build so maybe if someone can guess possibly-fragile packages for ICU version bump we should ofborg-build them explicitly.

About this, can we create some test suite, maybe in the form of a NixOS Test Module?

**Comment by 7c6f434c** [comment]
Whatever packages we want to be built, can be added to `passthru.tests` directly as packages. If you have a command-line check in mind, we can add it there as a `runCommandNoCC` expression. If we want to test something with a GUI, then yes, NixOS test might be needed…

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:21450]: ARM Darwin passed.

**Comment by AndersonTorres** [comment]
> Whatever packages we want to be built, can be added to passthru.tests directly as packages. 

This can be easy for some leaf packages like we do with live555-vlc.
But for things like, say, a statistical sample of 10 packages, I believe we should use something more structured.

**Comment by afh** [comment]
In regards to testing is there anything that you, @AndersonTorres and @7c6f434c, would like to see get added with this PR or should an issue be created that tracks the work?

**Comment by AndersonTorres** [comment]
This can be done later.  
For now let's work on make this PR merged before the end-of-year release :)

**Comment by 7c6f434c** [comment]
@ofborg build cfdg python3Packages.pyicu xfsprogs vte libical

**Comment by afh** [comment]
Unfortunately the failing ofborg builds failed due to a timeout trying to build llvm-16. Anything that could be done to address this?

`error: building of '/nix/store/03390naaq49frd957nhj31b61a33v8m9-llvm-16.0.6.drv^dev' from .drv file timed out after 3600 seconds`

**Comment by 7c6f434c** [comment]
Not really, Darwin + staging often ends up impossible to check on ofBorg due to stdenv being too heavy…

**Pull request merged**